src/pages/background/index.ts

```diff
@@ -40,5 +43,49 @@ browser.runtime.onMessage.addListener((msg : ExtMessage, sender) => {
       tabId: sender.tab.id,
       path: "assets/img/disabled.png"
     });
+  } else if (msg.command === 'corsUnblock') {
+    // CORS unblocking is restricted to the tab and page only for security reasons
+    console.log(`Unblocking ${msg.url} for ${msg.for} on tab ${sender.tab.id}`);
+    corsUnblockedUrls.push({
+      url: msg.url,
+      for: msg.for,
+      tab: sender.tab.id
+    });
   }
-});
\ No newline at end of file
+});
+
+// CORS Unblock
+browser.webRequest.onHeadersReceived.addListener((details : WebRequest.OnHeadersReceivedDetailsType) => {
+  // @ts-ignore
+  const initator = details.initiator;
+
+  if (
+    // Don't allow CORS unblocking anything other than audio and video
+    (details.type !== 'media' && details.type !== 'xmlhttprequest') ||
+
+    // Only remove CORS for GET and HEAD requests
+    (details.method !== 'GET' && details.method !== 'HEAD') ||
+
+    // Ignore requests that don't contain headers
+    (!details.responseHeaders) ||
+
+    // Don't allow CORS unblocking for requests that haven't been unblocked
+    // @ts-ignore
+    (corsUnblockedUrls.find((item) => item.url === details.url && item.for === initator && details.tabId === item.tab) === undefined)
+  ) {
+    return;
+  }
+
+  const header = details.responseHeaders.find(({name}) => name.toLowerCase() === 'access-control-allow-origin');
+  if (header) {
+    header.value = initator;
+  } else {
+    details.responseHeaders.push({name: "Access-Control-Allow-Origin", value: initator});
+  }
+
+  return {
+    responseHeaders: details.responseHeaders,
```

Comment:
This would be a security issue. If it is to be implemented, I think users at least need to know what they're signing up for.
https://webaudio.github.io/web-audio-api/#MediaElementAudioSourceOptions-security

Reply:
Yes, I agree. The current implementation is only for testing, and I'll probably be removing the complete CORS unblock mechanism as it doesn't really fix the "MediaElementAudioSource outputs zeroes due to CORS access restrictions" as I hoped it would